actors/eeoa/src/lib.rs

```diff
@@ -0,0 +1,136 @@
+use fvm_ipld_encoding::{Cbor, RawBytes};
+use fvm_shared::address::Payload;
+use fvm_shared::error::ExitCode;
+use fvm_shared::{MethodNum, METHOD_CONSTRUCTOR};
+use num_derive::FromPrimitive;
+use num_traits::FromPrimitive;
+use serde::Deserialize;
+use serde::Serialize;
+
+use fil_actors_runtime::runtime::{ActorCode, Runtime};
+use fil_actors_runtime::{
+    actor_error, restrict_internal_api, ActorError, AsActorError, EAM_ACTOR_ID, SYSTEM_ACTOR_ADDR,
+};
+
+#[cfg(feature = "fil-actor")]
+fil_actors_runtime::wasm_trampoline!(EeoaActor);
+
+/// Ethereum Externally Owned Address actor methods.
+#[derive(FromPrimitive)]
+#[repr(u64)]
+pub enum Method {
+    Constructor = METHOD_CONSTRUCTOR,
+}
+
+/// Ethereum Externally Owned Address actor state.
+#[derive(Default, Deserialize, Serialize)]
+#[serde(transparent)]
+pub struct State([(); 0]);
+
+impl Cbor for State {}
+
+/// Ethereum Externally Owned Address actor.
+pub struct EeoaActor;
+
+impl EeoaActor {
+    /// Ethereum Externally Owned Address actor constructor.
+    pub fn constructor(rt: &mut impl Runtime) -> Result<(), ActorError> {
+        rt.validate_immediate_caller_is(std::iter::once(&SYSTEM_ACTOR_ADDR))?;
+
+        let valid = rt
+            .lookup_address(rt.message().receiver().id().unwrap())
+            .map(
+                |a| matches!(a.payload(), Payload::Delegated(da) if da.namespace() == EAM_ACTOR_ID),
+            )
+            .with_context_code(ExitCode::USR_ILLEGAL_ARGUMENT, || {
+                "receiver must have predictable address".to_string()
+            })?;
+        if !valid {
+            return Err(ActorError::illegal_argument(
+                "invalid target for EEOA creation".to_string(),
+            ));
+        }
```

Comment:
There MUST be a Rust-ier way to do this, and @Stebalien I look forward to your pedantry.

Reply:
Honestly, I'd remove the `.map` and just use a single match. I.e.:

```rust
match rt.lookup_address(rt.message().receiver().id().unwrap()).map(|a|a.payload()) {
    Some(Payload::Delegated(da)) if da.namespace() == EAM_ACTOR_ID => {},
    Some(_) => return Err(ActorError::illegal_argument(
        "invalid target for EEOA creation".to_string(),
    )),
    None => return Err(ActorError::illegal_argument(
        "receiver must have a predictable address".to_string(),
     )),
}
```

Reply:
neat, i don't know about the conditional Some thingy...thanks!